go/tools/dokanclean/main.go

```diff
@@ -90,53 +89,66 @@ func findDokanUninstall(list bool, wow64 bool) (result int) {
 
 	k, err := registry.OpenKey(registry.LOCAL_MACHINE, "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall", access)
 	if err != nil {
-		fmt.Printf("Error %s opening uninstall subkeys\n", err.Error())
+		log.Printf("Error opening SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall : %s\n", err.Error())
 		return
 	}
-	defer k.Close()
 
-	names, err := k.ReadSubKeyNames(-1)
+	kb, err := registry.OpenKey(registry.CURRENT_USER, "SOFTWARE\\Keybase\\Keybase", access)
 	if err != nil {
-		fmt.Printf("Error %s reading subkeys\n", err.Error())
+		log.Printf("Error opening SOFTWARE\\Keybase\\Keybase:  %s\n", err.Error())
 		return
 	}
-	for _, name := range names {
-		subKey, err := registry.OpenKey(k, name, registry.QUERY_VALUE)
+	defer k.Close()
+	defer kb.Close()
+
+	var codes []string
+	dokanProductCode86, _, err := kb.GetStringValue("DOKANPRODUCT86")
+	if err == nil {
+		codes = append(codes, dokanProductCode86)
+	}
+
+	dokanProductCode64, _, err := kb.GetStringValue("DOKANPRODUCT64")
+	if err == nil {
+		codes = append(codes, dokanProductCode64)
+	}
+
+	for _, code := range codes {
+		subKey, err := registry.OpenKey(k, code, registry.QUERY_VALUE)
 		if err != nil {
-			fmt.Printf("Error %s opening subkey %s\n", err.Error(), name)
+			// This is not an error; we usually will find only one
+			// fmt.Printf("Error %s opening subkey %s\n", err.Error(), code)
+			continue
 		}
 
 		displayName, _, err := subKey.GetStringValue("DisplayName")
 		if list {
-			fmt.Printf("  %s -- %s\n", name, displayName)
+			log.Printf("  %s -- %s\n", code, displayName)
 		}
-		if err == nil && strings.HasPrefix(displayName, "Dokan Library") {
 
-			fmt.Printf("Found %s  %s\n", displayName, name)
-			uninstall, _, err := subKey.GetStringValue("QuietUninstallString")
-			if err != nil {
-				uninstall, _, err = subKey.GetStringValue("UninstallString")
-			}
-			if err != nil {
-				fmt.Printf("Error %s opening subkey UninstallString", err.Error())
-			} else {
-				result = mergeResults(result, doUninstallAction(uninstall, list))
-			}
+		uninstall, _, err := subKey.GetStringValue("QuietUninstallString")
+		if err != nil {
+			uninstall, _, err = subKey.GetStringValue("UninstallString")
+		}
+		if err != nil {
+			log.Printf("Error opening subkey UninstallString: %s", err.Error())
+		} else {
+			result = mergeResults(result, doUninstallAction(uninstall, list, log))
```

Comment:
That's how [the installer treats this key](https://github.com/keybase/client/blob/master/packaging/windows/WIXInstallers/KeybaseApps/KeybaseApps.wxs#L104), but I will double check